ospfd/ospf_zebra.c

```diff
@@ -1342,6 +1349,247 @@ void ospf_zebra_vrf_deregister(struct ospf *ospf)
 		zclient_send_dereg_requests(zclient, ospf->vrf_id);
 	}
 }
+
+/* Label Manager Functions */
+/**
+ * Request Label Range to the Label Manager.
+ *
+ * @param base		base label
+ * @param chunk_size	size of the label range
+ * @return		0 on success or -1 on failure
+ */
+int ospf_zebra_request_label_range(uint32_t base, uint32_t chunk_size)
+{
+	int ret;
+	uint32_t start, end;
+
+	if (zclient_sync->sock == -1)
+		ospf_zebra_label_manager_connect();
+
+	ret = lm_get_label_chunk(zclient_sync, 0, base, chunk_size, &start,
+				 &end);
+	if (ret < 0) {
+		zlog_warn("%s: error getting label range!", __func__);
+		return -1;
+	}
+
+	return 0;
+}
+
+/**
+ * Release Label Range to the Label Manager.
+ *
+ * @param start	Start of the previously allocated label range
+ * @param end	End of the previously allocated label range
+ */
+void ospf_zebra_release_label_range(uint32_t start, uint32_t end)
+{
+	int ret;
+
+	if (zclient_sync->sock == -1)
+		ospf_zebra_label_manager_connect();
+
+	ret = lm_release_label_chunk(zclient_sync, start, end);
+	if (ret < 0)
+		zlog_warn("%s: error releasing label range!", __func__);
+}
+
+/**
+ * Request a Label Chunk to the Label Manager. Label chunk has a fix size
+ * of 64 labels.
+ *
+ * @return	0 on success or -1 on failure
+ */
+static int ospf_zebra_get_label_chunk(void)
+{
+	int ret;
+	uint32_t start, end;
+	struct label_chunk *new_label_chunk;
+
+	if (zclient_sync->sock == -1)
+		ospf_zebra_label_manager_connect();
+
+	ret = lm_get_label_chunk(zclient_sync, 0, MPLS_LABEL_BASE_ANY,
+				 CHUNK_SIZE, &start, &end);
+	if (ret < 0) {
+		zlog_warn("%s: error getting label chunk!", __func__);
+		return -1;
+	}
+
+	new_label_chunk = calloc(1, sizeof(struct label_chunk));
+	if (!new_label_chunk) {
+		zlog_warn("%s: error trying to allocate label chunk %u - %u",
+			  __func__, start, end);
+		return -1;
+	}
+
+	new_label_chunk->start = start;
+	new_label_chunk->end = end;
+	new_label_chunk->used_mask = 0;
+
+	listnode_add(label_chunk_list, (void *)new_label_chunk);
```

Comment:
nit: unnecessary explicit cast to `void *`

Reply:
done